audio/filter/af_scaletempo.c

```diff
@@ -422,18 +423,20 @@ static bool reinit(struct mp_filter *f)
         memset(s->buf_overlap, 0, s->bytes_overlap);
         if (use_int) {
             int32_t *pb = s->table_blend;
-            int64_t blend = 0;
+            const float scale = M_PI / frames_overlap;
             for (int i = 0; i < frames_overlap; i++) {
-                int32_t v = blend / frames_overlap;
+                // Hann function
+                const int32_t v = 0.5f * (1.0f - cosf(i * scale)) * 65536 + 0.5;
                 for (int j = 0; j < nch; j++)
                     *pb++ = v;
-                blend += 65536; // 2^16
             }
             s->output_overlap = output_overlap_s16;
         } else {
             float *pb = s->table_blend;
+            const float scale = M_PI / frames_overlap;
             for (int i = 0; i < frames_overlap; i++) {
-                float v = i / (float)frames_overlap;
+                // Hann function
+                const float v = 0.5f * (1.0f - cosf(i * scale));
```

Comment:
I don't even remotely claim to understand the math here, but scaling this to `M_PI` makes the window function `0.0` at `i = 0` and `1.0` at `i = frames_overlap-1`. Maybe the intent was to use `2*M_PI`, leading to a window with a value of `0.0` towards either edge?

Reply:
I only need half the window function, because during the overlap one part of it does (1 - factor).